CollAction/Frontend/app/global/ShareButtons.tsx

```diff
@@ -14,98 +14,138 @@ interface IShareButtonsState {
-  }
+    async getLinkedInShareCount() {
+        try {
+            const apiLink: string = `https://www.linkedin.com/countserv/count/share?url=${this.getUrl()}&format=json`;
+            const response: Response = await fetch(apiLink);
+            const parsed = await response.json();
+            return parsed.count;
+        } catch (e) {
+            return 0;
+        }
+    }
 
+    async getFacebookShareCount() {
+        const apiLink: string = `http://graph.facebook.com/?id=${this.getUrl()}`;
+        const response: Response = await fetch(apiLink);
+        const parsed = await response.json();
+        return parsed.share.share_count;
+    }
 
+    async getShareCount() {
+        try {
+            const shareCount: number = await this.getFacebookShareCount();
+            this.setState({ shareCount, shareCountState: States.READY });
+        } catch (e) {
+            this.setState({ shareCountError: e, shareCountState: States.ERROR });
+        }
+    }
+
+    getTwitterUrl() {
+        return encodeURI(`https://twitter.com/intent/tweet?text=${this.getTitle()}&url=${this.getUrl()}`);
+    }
+
+    getFacebookUrl() {
+        return encodeURI(`https://www.facebook.com/sharer/sharer.php?u=${this.getUrl()}`);
+    }
+
+    getLinkedInUrl() {
+        return encodeURI(`http://www.linkedin.com/shareArticle?mini=true&url=${this.getUrl()}&title=${this.getTitle()}`);
+    }
+
+    getWhatsAppUrl() {
+        return encodeURI(`whatsapp://send>text=${this.getUrl()} ${this.getTitle()}`);
+    }
+
+    getShareButtonColWidth() {
+        if (this.state.width < 600) {
+            return "col-xs-2";
+        }
+        return "col-xs-3";
+    }
+
+    getShareLabelColWidth() {
+        if (this.state.width < 600) {
+            return "col-xs-4 share-count";
+        }
+        return "col-xs-3 share-count";
+    }
 
   render() {
-    return (
-      <div className="share-buttons">
-        <div className="row">
-          <div className="col-xs-3 share-count">
-            {this.state.shareCount}<br /> Shares
-          </div>
-          <div className="col-xs-3">
-            <a href={this.getFacebookUrl()} target="_blank">
-              <div className="social-media-share-buttons social-media-share-button-facebook">
-                <i className="fa fa-facebook"></i>
-              </div>
-            </a>
-          </div>
-          <div className="col-xs-3">
-            <a href={this.getTwitterUrl()}>
-              <div className="social-media-share-buttons social-media-share-button-twitter">
-                <i className="fa fa-twitter"></i>
-              </div>
-            </a>
-          </div>
-          <div className="col-xs-3">
-            <a href={this.getLinkedInUrl()}>
-              <div className="social-media-share-buttons social-media-share-button-linkedin">
-                <i className="fa fa-linkedin"></i>
+      return (
+          <div className="share-buttons">
```

Comment:
It's probably better to lay out the social media buttons in one bootstrap column, with a little margin. That way, you don't need the javascript-code to listen to the window size and determine wether you need xs-2 or xs-3.

Like this:

      <div className="share-buttons">
        <div className="row">
          <div className="col-xs-3 share-count">
            {this.state.shareCount}<br /> Shares
          </div>
          <div className="col-xs-9">
            <a href={this.getFacebookUrl()} target="_blank">
              <div className="social-media-share-buttons social-media-share-button-facebook">
                <i className="fa fa-facebook"></i>
              </div>
            </a>
            <a href={this.getTwitterUrl()}>
              <div className="social-media-share-buttons social-media-share-button-twitter">
                <i className="fa fa-twitter"></i>
              </div>
            </a>
            <a href={this.getLinkedInUrl()}>
              <div className="social-media-share-buttons social-media-share-button-linkedin">
                <i className="fa fa-linkedin"></i>
              </div>
            </a>
            <a href={this.getLinkedInUrl()}>
              <div className="social-media-share-buttons social-media-share-button-whatsapp">
                <i className="fa fa-whatsapp"></i>
              </div>
            </a>
          </div>
        </div>
      </div>

Then add this to the .social-media-share-buttons class:

    float: left;
    margin-right: 5px;

Reply:
Perhaps you can find a way to remove the margin-right: 5px for the last button; I had some trouble figuring that out.